components/camel-aws/camel-aws2-kinesis/src/main/java/org/apache/camel/component/aws2/kinesis/Kinesis2Endpoint.java

```diff
@@ -99,12 +101,20 @@ public Producer createProducer() throws Exception {
 
     @Override
     public Consumer createConsumer(Processor processor) throws Exception {
-        final Kinesis2Consumer consumer = new Kinesis2Consumer(this, processor);
+        var kinesisConnection = KinesisConnection.getInstance();
+        final Kinesis2Consumer consumer = new Kinesis2Consumer(this, processor, kinesisConnection);
         consumer.setSchedulerProperties(getSchedulerProperties());
+        startHealthChecks(kinesisConnection);
         configureConsumer(consumer);
         return consumer;
     }
 
+    private void startHealthChecks(KinesisConnection kinesisConnection) {
```

Comment:
For this you could use the executor service we have in Camel, like we did here: 

https://github.com/apache/camel/blob/main/components/camel-aws/camel-aws2-s3/src/main/java/org/apache/camel/component/aws2/s3/stream/AWS2S3StreamUploadProducer.java#L83

Reply:
What a great idea, I've changed the code according to your suggestion

Reply:
@oscerd  it's ready to go, I suppose

Reply:
@Andrea Cosentino Please retrigger the pipeline one more time